cicd/build.sh

```diff
@@ -40,28 +45,34 @@ function docker_build {
         {
             set -x
             docker pull "${IMAGE}" &&
-            docker build -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/$DOCKERFILE --cache-from "${IMAGE}"
+            docker build $IMAGE_TAG_OPTS $APP_ROOT -f $APP_ROOT/$DOCKERFILE --cache-from "${IMAGE}"
             set +x
         } || {
             echo "Build from cache failed, attempting build without cache"
             set -x
-            docker build -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/$DOCKERFILE
+            docker build $IMAGE_TAG_OPTS $APP_ROOT -f $APP_ROOT/$DOCKERFILE
             set +x
         }
     else
         set -x
-        docker build -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/$DOCKERFILE
+        docker build $IMAGE_TAG_OPTS $APP_ROOT -f $APP_ROOT/$DOCKERFILE
         set +x
     fi
     set -x
-    docker push "${IMAGE}:${IMAGE_TAG}"
+
+    docker push --all-tags "${IMAGE}:${IMAGE_TAG}"
```

Comment:
Question here -- does  `docker push --all-tags <image>:<image_tag>` push all tags that are ONLY the same as `image_tag` ? The only time I have seen it used is with syntax: `docker push --all-tags <image>` which pushes up ALL tags for the image.

If we aren't sure, should we just change this to use the same logic as the podman push below (using the `if`?)

Reply:
All tags for the image - I might be making a bad assumption here that the build machine would only have relevant tags for the image.